test/xds/xds_client_federation_test.go

```diff
@@ -288,9 +288,9 @@ func (s) TestFederation_UnknownAuthorityInDialTarget(t *testing.T) {
 	target = fmt.Sprintf("xds://unknown-authority/%s", serviceName)
 	t.Logf("Dialing target %q with unknown authority which is expected to fail", target)
 	wantErr := fmt.Sprintf("authority \"unknown-authority\" specified in dial target %q is not found in the bootstrap file", target)
-	_, err = grpc.Dial(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(xdsResolver))
+	_, err = grpc.NewClient(target, grpc.WithTransportCredentials(insecure.NewCredentials()), grpc.WithResolvers(xdsResolver))
 	if err == nil || !strings.Contains(err.Error(), wantErr) {
```

Comment:
`Dial` attempts to create a transport immediately, and would fail here. `NewClient` is lazy, so it will not fail immediately. To see the failure, do the following:
1. Call cc.Connect().
2. Wait for the clientConn to transition to TransientFailure using testutils.AwaitState.
3. Make an rpc using `_, err := client.EmptyCall(ctx, &testpb.Empty{})`.

Now you can do the check on the error returned from step 3.

Reply:
done